### ISSUE TITLE
PHP 8 compatibility

### ISSUE THREAD

**Issue by scottshipman** [RusticiSoftware/TinCanPHP#108]
the packagist / composer instance tops out at php 7.

Are there plans to be compatible with php 8?

**Comment by jakegibs617** [comment]
agreed, this is a major showstopper for me.

rusticisoftware/tincan requires php ~5.5 || ^7.0.3 -> your php version (8.1.0) does not satisfy that requirement.

**Comment by chippyjacob1990** [comment]
Yes this is very urgent for the project to migrate to php8. See  https://www.drupal.org/project/opigno_lms/issues/3269096

**Comment by bob-hinrichs** [comment]
Too bad this is still hanging! Php 7 is very soon to be done.

**Comment by mattpaz** [comment]
@brianjmiller - any thoughts on getting this committed (or insight into why it isn't)?

**Comment by tc60045** [comment]
any workarounds? fork time?

**Comment by bob-hinrichs** [comment]
in 18 days php 7.4 support ends.

**Comment by tc60045** [comment]
Thanks.  I swapped horses, moving from Drupal / Opigno to Wordpress / Sensei LMS.

**Comment by hamletta** [comment]
If you're stuck, the devs behind Opigno, a Drupal-based LMS, forked this package to solve the immediate problem: https://github.com/opigno/TinCanPHP

**Comment by mattpaz** [comment]
@alexrayu @spheresh 

Does the `willdurand/negotiation` dependency need to be bumped from `2.3.1` to `3.1.0`?

V3 added PHP 8 support.
https://github.com/willdurand/Negotiation/releases

**Comment by mattpaz** [comment]
I noticed that https://github.com/RusticiSoftware/TinCanPHP/pull/110  was committed, which is great.

But there still seems to be some work to do relative to PHP 8.1 support.
In addition to the `willdurand/negotiation` question, I observed at least one deprecated technique was in play:

```
Deprecated function: http_build_query(): 
Passing null to parameter #2 ($numeric_prefix) of type string is deprecated 
in TinCan\RemoteLRS->sendRequest() 
```

https://github.com/RusticiSoftware/TinCanPHP/blob/9758b3ec08653f7a49c8ab14ebd1f01557d89b78/src/RemoteLRS.php
